app/src/display/widgets/layer_status.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/display/widgets/layer_status.h>
+#include <zmk/events/layer-state-changed.h>
+#include <zmk/event-manager.h>
+#include <zmk/endpoints.h>
+#include <zmk/keymap.h>
+
+static sys_slist_t widgets = SYS_SLIST_STATIC_INIT(&widgets);
+static lv_style_t label_style;
+
+void layer_status_init() {
+    if (label_style.text.font != NULL) {
+        return;
+    }
+
+    lv_style_copy(&label_style, &lv_style_plain);
+    label_style.text.color = LV_COLOR_BLACK;
+    label_style.text.font = &lv_font_roboto_12;
+    label_style.text.letter_space = 1;
+    label_style.text.line_space = 1;
+}
+
+void set_layer_symbol(lv_obj_t *label) {
+    int active_layer_index = zmk_keymap_highest_layer_active();
+    char text[6] = {};
+
+    LOG_DBG("Layer changed to %i", active_layer_index);
+    sprintf(text, LV_SYMBOL_KEYBOARD "%i ", active_layer_index);
```

Comment:
Future: We'll want to use the (optional) layer label for this, for nicer display.

Reply:
I'm not sure I follow what changes may need to made for this item, can you clarify?

Reply:
None for this PR.

Eventually, I'd like each layer to get a label in the DT:

```
    nav_layer {
        label = "NAV";
        bindings = <>;
   };
```

And then the display code would show that label, not just the numeric layer # value.

Reply:
Understood ; Does that need to be implemented as part of this PR / should this particular edit or comment be closed?

Reply:
This comment can be closed. Just a note for next/future work, doesn't need to be this PR.